### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by sgolubev33** [HCL-TECH-SOFTWARE/domino-backup#4]
**HCL Product Version**
Domino 12.0.1FP1

**Describe the bug**
Domino Backup and Veeam Backup & Replication integration error.

 I've configured backup for Domino on Linux on the following platforms:

    CentOS 8 Stream. Link to download log files (IBM_TECHNICAL_SUPPORT folder and /local/backup/log): https://cyone.box.com/s/mbn8g7wfz7e06ey4ckwzo8p6epmtp5ma
    Astra Linux CE 2.12.45 (orel). Link to download log files(IBM_TECHNICAL_SUPPORT folder and /local/backup/log): https://cyone.box.com/s/ncb8tuvo0u8eihw8tv4vkv60taz4dq3q

On both platforms I receiving the same error while running the pre-freeze script:
[006819:000002-00007EFE9669E740] 11/08/2022 07:34:06 PM  Backup: Error executing Snapshot Backup Command: [ ERROR - No Snapshot status. Timeout reached] : File cannot be created

Virtualization platform: VmWare vSphere 6.7

Veeam Backup & Replication version 11.0.1.1261

HCL Case Number: CS0354719

**Comment by Daniel-Nashed** [comment]
Thanks for the question. I would not call it a bug yet. It's probably a configuration issue.
I can see that the snapshot backup is triggered. But the snapshot command returns an error.

[General ERROR!] - Error executing Snapshot Command 

Domino 12.0.1 introduced end to end logging. You should see an error log in the backup log document in that case.

The snapshot command is the command, which reports the status back to the initiating script of the backup waiting to finish. 

Can you check if you find more logging and can you double check if all scripts are correctly configured and match the configuration?

**Comment by sgolubev33** [comment]
The error log of the backup job contains only the following lines:
----------------------------------------
/opt/hcl/domino/backup/veeam/backup_snapshot.sh 2>&1 "" "/local/notesdata" "" "astra12" "default" "SNAP" "20221109175122" "/local/backup/nsf" "7" "20221116175122"
----------------------------------------
Removed tag file [/local/notesdata/dominobackup_20221109175122.tag]
Return: ERROR - No Snapshot status. Timeout reached
----------------------------------------

**Comment by Daniel-Nashed** [comment]
We synced up directly and there have been multiple edge cases we ran into.

The first one was that user used to invoke the freeze script was the root user. The root user creates the lock file handling the status for the backup and, the Domino server's user "**notes**"" was not allowed to write it.

We are going to fix that by changing the owner if this file to the notes user, when running the scripts with the root user.
But clearly we want the script executed **not elevated** for security reasons and use the standard notes user instead.

Along the way during the restore we ran into some other parts that could be sanitized:

- The way we look for the user in Veeam should be checked with `-eq` instead of `like` in the powershell script.
- We will also improve the logging for the Linux scripts
- Add more documentation to the scripts and their purpose

There is a tracefile configuration, which helped us to find the details.

```
# Log and trace files
#TRACEFILE=/tmp/dominoveeam_trace.log
```
The status file is found here:

```
DOMBACK_STATUS_FILE=$DOMINO_DATA_PATH/dominobackup_snapshot.lck
```

**Comment by Daniel-Nashed** [comment]
@TobiasGrosse , you are using the same scripts for a different integration with no detailed information. How should we help in this case? 

The script has debug functionality and I added information to this issue what we ran into here.
I am assuming you are re-using the existing Veeam scripts for your own integration? 

Please look into the script files and enable the logging. Also it would make sense to look into your specific backup application to see if we have a better way to start the snap shot and build an easier integration that have the backup triggered by Domino. 
we are using the trigger from the Veeam side, because this is the easiest flow they support in this context.
Maybe Rubrik has a way to trigger a snapshot directly?

If there is no way, you could use the same flow, we implemented with Veeam. 
But I would check first, if there is an easier integration point. I have never worked with  Rubrik and  I would be interested to help. Sadly we have no access to  Rubrik. I have looked into their command reference earlier and it looks like we would find an easier integration point.

**Comment by Daniel-Nashed** [comment]
looking into your logs I see that you are not exactly in the same status.
Your status is that the Domino side is done `DOMINO-DONE` and the backup application would need to get the trigger the snapshot can be taken. the next step in the flow would be, that the backup sends an unfreeze trigger to Domino backup.
So your situation is not the same, than the issue we had in this thread.

**Comment by TobiasGrosse** [comment]
Hi @Daniel-Nashed 
thanks for the quick response.
AFAIK Rubrik can trigger a snapshot directly. Maybe we could discuss this together with the customer, as I believe the scripts might not be required...
I have a similar (working) setup with veeam using the scripts.

**Comment by Daniel-Nashed** [comment]
Great to hear that Veeam works for you!

For Veeam on Linux in Domino 12.0.2 you will continue to use this flow.
But the just released version Domino 12.0.2 on Windows, we recommend to use the new VSS Writer integration.
I rewrote the documentation for the [Veam integration on Windows](https://opensource.hcltechsw.com/domino-backup/backup-providers/veeam/install_vss_writer/). So in case you are also using Windows, please have a look into the new VSS Writer integration,where the flow uses VSS as a trigger without any freeze scripts involved.
